src/video/x11/SDL_x11xinput2.c

```diff
@@ -164,14 +164,86 @@ X11_InitXinput2(_THIS)
 #endif
 }
 
+
+/* 
+    The following two functions should just return whether or wheter not the setting "Natural scrolling" or "Reverse scroll direction" is set.
+    This should be working on all major desktop environments.
+*/
+
+static bool X11_IsMousepadReversed(void)
+{
+    return false;
+}
+
+static bool X11_IsMouseReversed(void)
+{
+    return false;
+}
+
+bool X11_IsXinput2DeviceMousepad(Display *display, int sourceId)
+{
+    XIDeviceInfo *info;
+    int nDevices;
+    bool isMousepad = false;
+
+    info = X11_XIQueryDevice(display, sourceId, &nDevices);
+    
+    /* Is this the best way to do it? Probably not. Maybe we can check some device properties, but for now this should work aswell. */
+    if (strstr(info[0].name, "Synaptics TouchPad") != NULL) {
```

Comment:
Definitely not, there are lots and lots of different mouse pads out there.

Reply:
Also, this is an expensive query to make every time a button is pressed.

Reply:
Yeah, I know. This isn't supposed to be a final version, rather a prototype. My idea was to handle hierarchy changes and do these checks there, and only once. This would save much computational power. Then we would have an active list of all slave devices and their mousepad/touchpad status. I think I would rename everything to touchpad, as it's the appropriate name. I just wanted to publish something, so we can get into implementation details chat.

Reply:
Sounds good.